torch/csrc/jit/codegen/cuda/arith.cpp

```diff
@@ -975,12 +975,23 @@ TensorView* expand(TensorView* inp, const std::vector<Val*>& expanded_sizes) {
   auto inp_domain = TensorDomain::noReductions(inp->getMaybeRFactorDomain());
 
   TORCH_CHECK(
-      expanded_sizes.size() == inp_domain.size(),
-      "Invalid expand, number of sizes provided is expected to be ",
+      expanded_sizes.size() >= inp_domain.size(),
+      "Invalid expand, number of sizes provided is expected to be at least ",
       inp_domain.size(),
       " but received ",
       expanded_sizes.size());
 
+  // broadcast inner on inp to match rank with other.
+  if (inp_domain.size() < expanded_sizes.size()) {
+    const int num_bcast =
+        static_cast<int>(expanded_sizes.size() - inp_domain.size());
+    std::vector<bool> inner_bcast_dims(expanded_sizes.size(), false);
+    std::fill(
+        inner_bcast_dims.begin(), inner_bcast_dims.begin() + num_bcast, true);
```

Comment:
This inserts broadcast axes to outside of inp_domain, i.e., left of the input domains. Is it what you intend to do?

Reply:
Yes it is. See note in PR (could use better English :stuck_out_tongue: ): `updated expand(_as) in arith.cpp to allow rank broadcast on inner dimensions. (matching pytorch API).`

```
In [1]: import torch

In [2]: x = torch.randn(5, 5)

In [3]: x.expand((2, 5, 5)).shape
Out[3]: torch.Size([2, 5, 5])

```

Reply:
I can back this out into integration code if we think it's a crazy API tho.

Reply:
I don't see any problem, so no need to back it out.